crates/sui-adapter/src/adapter.rs

```diff
@@ -77,9 +78,25 @@ pub fn execute<
             CallArg::Pure(_) => None,
             CallArg::Object(ObjectArg::ImmOrOwnedObject((id, _, _)))
             | CallArg::Object(ObjectArg::SharedObject(id)) => {
-                Some((*id, state_view.read_object(id)?))
+                Some(vec![(*id, state_view.read_object(id)?)])
+            }
+            CallArg::ObjVec(vec) => {
```

Comment:
This is not an issue we need to solve now, but we're specializing the `CallArg` types here and I fear we might need to do it in future for other types.
Not sure but is it even possible to create a general container type that prevents over-specialization of the adapter and sui types?
So we can decouple the inner Move type impl from sui types.

Reply:
I share your concerns, but I am failing to think of anything that much more general.

I have calmed myself with the following line of reasoning:
* We can safely add new variants to this if needed, it will not be a breaking change (just will need to be epoch gated or gated by some similar mechanism)
* With that if we have something new in the future, like a map or something, we can do that
* Or similarly, if we come up with the supreme general approach, we can add that later while still supporting this special case (we did that with Move args in Diem fwiw. As we did not originally support general BCS arguments, only a few specific hand rolled cases)
* We can always interpret this differently in the adapter. Much like `CallArg::Object` can be used to populate `&Obj`, `&mut Obj` and `Obj`, we can have more than one meaning for `ObjVec` 

With this it feels fine enough to just be specific for vectors right now.